### PR TITLE
Port DeferredCredits test to jtx and views

### PR DESCRIPTION
**Pull request by seelabs**

### CODE REVIEW

**Comment by seelabs** [comment]
@HowardHinnant @miguelportilla All the commits except `Account for deferred credits of parent views` and `Port DeferredCredits test to jtx and views` are part of other PRs and do not need to be reviewed.

**Comment by HowardHinnant** [comment]
Needed to compile on OS X:
https://github.com/HowardHinnant/rippled/commits/dc-test

**Comment by seelabs** [comment]
@HowardHinnant Thanks. Cherry picked your commit and testing on all platforms now (as I should have done in the first place).

**Comment by HowardHinnant** [comment]
:+1:

**Comment by vinniefalco** [comment]
You can rebase against `view` again, its been squashed

**Comment by miguelportilla** [comment]
:+1:

**Comment by vinniefalco** [comment]
This can be rebased against the `view` branch

**Comment by seelabs** [comment]
Squashed and rebased against develop. I also made a trivial change addressing @HowardHinnant style guide issue.

File: src/ripple/app/ledger/tests/DeferredCredits.test.cpp [hunk 1]
```diff
@@ -55,228 +56,198 @@ class DeferredCredits_test : public beast::unit_test::suite
...
-        auto& gw1 = accounts["gw1"];
-        auto& gw2 = accounts["gw2"];
-        auto& snd = accounts["snd"];
-        auto& rcv = accounts["rcv"];
+        using namespace jtx;
+        Env env (*this);
+        Account const gw1 ("gw1");
+        Account const gw2 ("gw2");
+        Account const snd ("snd");
+        Account const rcv ("rcv");
```
Comment (vinniefalco) [inline src/ripple/app/ledger/tests/DeferredCredits.test.cpp]: All members of `Account` are const. Putting `const` on the local doesn't hurt. But it also doesn't help.

**Comment by HowardHinnant** [comment]
Needs trivial rebase to b6.

**Comment by seelabs** [comment]
Rebased to b6 and pushed, but I'm getting compile errors. Will fix an push again shortly.

**Pull request merged**